### ISSUE TITLE
Memory and descriptor leaks in Remoted

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#2380]
|Version|Revision|
|---|---|
|3.8|3802|

Testing key address integrity hash table:
- Client keys length: 12446 entries.
- Key reloading interval: 1 second.
- Runtime: ~one minute.
- Key address integrity hash table length: ~217K entries.
- False positives (address collisions): ~38K.
- File descriptor peak number: 50.000.

## Memory leak

```
==17940== 7,533 (4,608 direct, 2,925 indirect) bytes in 32 blocks are definitely lost in loss record 56 of 96
==17940==    at 0x4C2DBC5: calloc (vg_replace_malloc.c:711)
==17940==    by 0x168223: OS_DupKeyEntry (keys.c:647)
==17940==    by 0x168109: OS_DupKeys (keys.c:629)
==17940==    by 0x167894: OS_UpdateKeys (keys.c:453)
==17940==    by 0x115EFC: check_keyupdate (sendmsg.c:44)
==17940==    by 0x119F40: rem_keyupdate_main (secure.c:268)
==17940==    by 0x5A46493: start_thread (pthread_create.c:333)
==17940==
==17940== 993,420 (605,088 direct, 388,332 indirect) bytes in 4,202 blocks are definitely lost in loss record 93 of 96
==17940==    at 0x4C2DBC5: calloc (vg_replace_malloc.c:711)
==17940==    by 0x1663AF: OS_AddKey (keys.c:75)
==17940==    by 0x16721D: OS_ReadKeys (keys.c:302)
==17940==    by 0x16793F: OS_UpdateKeys (keys.c:461)
==17940==    by 0x115EFC: check_keyupdate (sendmsg.c:44)
==17940==    by 0x119F40: rem_keyupdate_main (secure.c:268)
==17940==    by 0x5A46493: start_thread (pthread_create.c:333)
==17940==
```

## Crashing error
```
2019/01/16 22:01:26 ossec-remoted: ERROR: Unable to open agent file. errno: 23
2019/01/16 22:01:26 ossec-remoted: CRITICAL: (1103): Could not open file '/queue/rids/11579' due to [(23)-(Too many open files in system)].
```

**Comment by vikman90** [comment]
We found an extra memory leak at shared file management:

```
==9597== 16 bytes in 1 blocks are definitely lost in loss record 16 of 67
==9597==    at 0x4C2DBC5: calloc (vg_replace_malloc.c:711)
==9597==    by 0x112E48: c_files (manager.c:641)
==9597==    by 0x11526B: manager_init (manager.c:1262)
==9597==    by 0x119496: HandleSecure (secure.c:61)
==9597==    by 0x11AFF9: HandleRemote (remoted.c:97)
==9597==    by 0x11B792: main (main.c:215)
==9597==
==9597== 256 bytes in 16 blocks are definitely lost in loss record 36 of 67
==9597==    at 0x4C2DBC5: calloc (vg_replace_malloc.c:711)
==9597==    by 0x112E48: c_files (manager.c:641)
==9597==    by 0x114DAB: update_shared_files (manager.c:1159)
==9597==    by 0x5A46493: start_thread (pthread_create.c:333)
```

**Comment by chemamartinez** [comment]
Merged at #2381. Thanks @vikman90!
